src/components/features/comments/comments.js

```diff
@@ -0,0 +1,602 @@
+ * @returns {HTMLElement} The rendered comment element
+ */
+function renderComment({ comment, projectId, onCommentAction, api, page }) {
+  const user = authService.getCurrentUser();
+
+  const isDeleted =
+    comment.isDeleted || comment.message === "[deleted]" || !comment.User;
+
+  const el = createEl("div", {
+    class: isDeleted ? "comment deleted" : "comment",
+    "data-comment-id": comment.id,
+  });
+
+  if (isDeleted) {
+    const metaRow = createEl("div", { class: "comment-meta" });
+    const userSection = createEl("div");
+    const deletedLabel = createEl(
+      "span",
+      { class: "comment-username" },
+      "Condolence deleted",
+    );
+    const time = createEl(
+      "span",
+      { class: "comment-time" },
+      timeAgo(comment.createdAt),
+    );
+
+    userSection.append(deletedLabel);
+    metaRow.append(userSection, time);
+
+    const msg = createEl("div", { class: "comment-message" }, "[deleted]");
+    el.append(metaRow, msg);
+
+    if (Array.isArray(comment.replies) && comment.replies.length > 0) {
+      const repliesContainer = createEl("div", { class: "comment-replies" });
+
+      comment.replies.forEach((reply) => {
+        repliesContainer.append(
+          renderComment({
+            comment: reply,
+            projectId,
+            onCommentAction,
+            api,
+            page,
+          }),
+        );
+      });
+
+      el.append(repliesContainer);
+    }
+
+    return el;
+  }
+
+  const initialsName =
+    comment.User?.fullName || comment.User?.username || "User";
+  const avatarUrl =
+    comment.User?.avatarUrl ||
+    `https://ui-avatars.com/api/?name=${encodeURIComponent(initialsName)}&background=random`;
+
+  const avatar = createEl("img", {
+    class: "comment-avatar",
+    src: avatarUrl,
+    alt: comment.User?.username || "User avatar",
+  });
+
+  const userName = createEl(
+    "span",
+    { class: "comment-username" },
+    comment.User?.username || "Anonymous",
+  );
+
+  const time = createEl(
+    "span",
+    { class: "comment-time" },
+    timeAgo(comment.createdAt),
+  );
+
+  const userSection = createEl("div");
+  userSection.append(avatar, userName);
+
+  const metaRow = createEl("div", { class: "comment-meta" });
+  metaRow.append(userSection, time);
+
+  let msg = createEl("div", { class: "comment-message" }, comment.message);
+  const actions = createEl("div", { class: "comment-actions" });
+  const leftActions = createEl("div");
+  const rightActions = createEl("div");
+
+  //Reply button (only for root comments and authenticated users)
+  if (user && user.id && comment.parentId === null) {
+    const replyBtn = createEl("button", { class: "btn-small" }, "Reply");
+    replyBtn.onclick = () => {
+      closeAllCommentForms();
+      if (!el.querySelector(".reply-form")) {
+        const form = renderReplyForm({
+          projectId,
+          parentId: comment.id,
+          api,
+        });
+        el.append(form);
```

Comment:
Perhaps prepend the form so it appears at the top just below the parent comment when replying to a comment that has lots of replies already, may user dislike having their view port moved.